### PR TITLE
VIH-9120 - Add KEDA MI to Service Bus

### PR DESCRIPTION
**Pull request by v1ferrarij**
### JIRA link (if applicable) ###

VIH-9120

### Change description ###

add keda MI to Service Bus

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x  ] No
```

### CODE REVIEW

File: terraform/09-video-hearing-core/modules/VHDataServices/main.tf [hunk 1]
```diff
@@ -158,3 +158,23 @@ resource "azurerm_servicebus_queue" "vh-infra-core" {
...
+  count               = local.environment == "dev" ? 0 : 1
+  name                = "keda-${local.environment}-mi"
+  resource_group_name = "managed-identities-${local.environment}-rg"
+}
+
+resource "azurerm_role_assignment" "Azure_Service_Bus_Data_Receiver" {
+  count                = local.environment == "dev" ? 0 : 1
+  scope                = azurerm_servicebus_namespace.vh-infra-core.id
+  role_definition_name = "Azure Service Bus Data Receiver"
+  principal_id         = data.azurerm_user_assigned_identity.keda_mi[0].principal_id
```
Comment (chris-pateman) [inline terraform/09-video-hearing-core/modules/VHDataServices/main.tf:172]: Could you used the IF statement here?

File: terraform/09-video-hearing-core/modules/VHDataServices/main.tf [hunk 2]
```diff
@@ -158,3 +158,23 @@ resource "azurerm_servicebus_queue" "vh-infra-core" {
...
+}
+
+resource "azurerm_role_assignment" "Azure_Service_Bus_Data_Receiver_Dev" {
+  count                = local.environment == "dev" ? 1 : 0
+  scope                = azurerm_servicebus_namespace.vh-infra-core.id
+  role_definition_name = "Azure Service Bus Data Receiver"
+  principal_id         = "8e65726d-ee0f-46e7-9105-f97ab9f5e70b" # Dev uses Staging MI.
```
Comment (chris-pateman) [inline terraform/09-video-hearing-core/modules/VHDataServices/main.tf:179]: Why does this need to be hard coded?
What MI is it using?

File: terraform/09-video-hearing-core/modules/VHDataServices/main.tf [hunk 3]
```diff
@@ -158,3 +158,23 @@ resource "azurerm_servicebus_queue" "vh-infra-core" {
   lock_duration         = "PT5M"
   max_size_in_megabytes = 1024
 }
+
+data "azurerm_user_assigned_identity" "keda_mi" {
```
Comment (chris-pateman) [inline terraform/09-video-hearing-core/modules/VHDataServices/main.tf:162]: why don't we need one for dev?

**Review by chris-pateman — CHANGES_REQUESTED** [review]

File: terraform/09-video-hearing-core/modules/VHDataServices/main.tf (hunk 2 above)
Comment (v1ferrarij) [reply terraform/09-video-hearing-core/modules/VHDataServices/main.tf:179]: Dev uses staging MI for some reason

File: terraform/09-video-hearing-core/modules/VHDataServices/main.tf (hunk 3 above)
Comment (v1ferrarij) [reply terraform/09-video-hearing-core/modules/VHDataServices/main.tf:162]: Dev uses staging MI for KEDA

**Review by chris-pateman — APPROVED** [review]
LGTM

**Pull request merged**